src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -477,6 +477,14 @@ protected function forward(Request $request, $catch = false, Response $entry = n
             $request->headers->set('X-Forwarded-For', $request->server->get('REMOTE_ADDR'));
         }
 
+        // modify the Forwarded header if needed
+        $forwarded = $request->headers->get('Forwarded');
+        if ($forwarded) {
+            $request->headers->set('Forwarded', preg_replace('/for=([0-9a-zA-Z.,\\s]*)/', '$0, '.$request->server->get('REMOTE_ADDR'), $forwarded));
```

Comment:
@nicolas-grekas , preg_replace because you can have this in the Froward header : proto=http; for=10.0.0.2, 10.0.0.3; by=203.0.113.43. Adding a new IP in for= with preg_replace is not the good way ? Also to deal the 'no for', you have the else condition. Maybe the condition is wrong according to you ? Shoud I add a test to check the existing word 'for' with strpos if the if condition ?